src/Shared/BinaryTranslator.cs

```diff
@@ -660,12 +660,31 @@ public void TranslateDictionary<T>(ref Dictionary<string, T> dictionary, IEquali
                 }
             }
 
-            /// <summary>
-            /// Reads in the boolean which says if this object is null or not.
-            /// </summary>
-            /// <typeparam name="T">The type of object to test.</typeparam>
-            /// <returns>True if the object should be read, false otherwise.</returns>
-            public bool TranslateNullable<T>(T value)
+            public void TranslateDictionary(ref Dictionary<string, DateTime> dictionary, StringComparer comparer)
+            {
+                if (!TranslateNullable(dictionary))
+                {
+                    return;
+                }
+
+                int count = _reader.ReadInt32();
+                dictionary = new(count, comparer);
+                string key = string.Empty;
+                DateTime val = DateTime.MinValue;
+                for (int i = 0; i < count; i++)
+                {
+                    Translate(ref key);
+                    Translate(ref val);
```

Comment:
as an aside, what happens if you build, change timezones, and build again? Do we potentially under build or do we handle that?

Reply:
I believe we underbuild, but that's very unusual and perpendicular to this PR. (Does that sound right @rainersigwald?) This part is just about serializing and deserializing something.

Reply:
Never mind - I'm pretty sure the DateTimes are all UTC so time zones can generally be ignored. DateTime barely has a concept of time zone anyway.